cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java

```diff
@@ -188,6 +188,18 @@ private void doTestToFromOldId(EntityId id) {
   }
 
   @Test
+  public void testDatasetName() {
+    // TODO: CDAP-5730: Even though dots are allowed, if we use it here, id creation will fail.
```

Comment:
@bdmogal You mean 'correct' all the shortcomings? Ya, depends on where all this class is being used. We have to do some not-so-simple parsing if we still want to support '.'s.